### PR TITLE
enh(a11y): Add accessible user menu semantics

### PR DESCRIPTION
**Pull request by Pytal**
- For https://github.com/nextcloud/server/issues/37095

## Summary

- Wrap menu trigger button in nav container
- Add distinct label to app menu navigation landmark
- Add user status to accessible name i.e. "Avatar of John — Online — 📅 — In a meeting"
- Test with NVDA

> No visual changes

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- Screenshots before/after for front-end changes
- Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by szaimen** [comment]
Sorry, dont feel able to review this

File: core/src/components/AppMenu.vue [hunk 1]
```diff
@@ -20,7 +20,8 @@
   -->
 
 <template>
-	<nav class="app-menu">
+	<nav class="app-menu"
+		:aria-label="t('core', 'App menu')">
```
Comment (ShGKme) [inline core/src/components/AppMenu.vue:24]: Not sure if "app menu" is understandable

```suggestion
	<nav class="app-menu" :aria-label="t('core', 'Applications menu')">
```

**Review by susnux — APPROVED** [review]
Agree the the note of @ShGKme but otherwise looks fine

**Review by JuliaKirschenheuter — APPROVED** [review]
looks good, thank you!

**Comment by Pytal** [comment]
Any objections @ShGKme?

**Comment by Pytal** [comment]
Otherwise we're good to go

**Comment by Pytal** [comment]
CI failures unrelated

**Pull request merged**